cross-chain/solana/programs/tbtc/src/lib.rs

```diff
@@ -0,0 +1,357 @@
+use anchor_lang::prelude::*;
+use anchor_spl::{
+    associated_token::AssociatedToken,
+    // mint,
+    token::{mint_to, TokenAccount, Mint as SplMint, MintTo as SplMintTo, Token as SplToken},
+};
+
+declare_id!("HksEtDgsXJV1BqcuhzbLRTmXp5gHgHJktieJCtQd3pG");
+
+#[program]
+pub mod tbtc {
+    use super::*;
+    
+    pub fn initialize(ctx: Context<Initialize>) -> Result<()> {
+        let tbtc = &mut ctx.accounts.tbtc; 
+        tbtc.authority = ctx.accounts.authority.key();
+        tbtc.token_mint = ctx.accounts.tbtc_mint.key();
+        tbtc.token_bump = *ctx.bumps.get("tbtc_mint").unwrap();
+        tbtc.minters = 0;
+        tbtc.guardians = 0;
+        tbtc.paused = false;
+        Ok(())
+    }
+
+    pub fn add_minter(ctx: Context<AddMinter>) -> Result<()> {
+        let minter_info = &mut ctx.accounts.minter_info;
+        minter_info.minter = ctx.accounts.minter.key();
+        minter_info.bump = *ctx.bumps.get("minter_info").unwrap();
+
+        ctx.accounts.tbtc.minters += 1;
+        emit!(MinterAdded { minter: ctx.accounts.minter.key() });
```

Comment:
Keep in mind that validators are not required to persist program logs (this is the equivalent of `msg!`) so if you want a history of when these events happen, you should store this information in an account.